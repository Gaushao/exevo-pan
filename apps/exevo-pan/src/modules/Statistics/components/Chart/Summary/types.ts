export interface SummaryProps extends React.HTMLAttributes<HTMLDivElement> {
  title: string
  value: number
  percentage: number
  positive?: boolean
}
