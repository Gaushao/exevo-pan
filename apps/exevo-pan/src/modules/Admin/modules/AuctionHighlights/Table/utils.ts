import {
  readableCurrentDate,
  ddmmyyy2mmddyyyy,
  sanitizeDateTime,
  dateToReadableStringDate,
  DAYS_IN,
  MILLISECONDS_IN,
} from 'utils'
import { HighlightStatus } from './types'

export const isPastDate = (readableDate: string): boolean => {
  const currentDate = sanitizeDateTime(new Date())
  const checkingDate = new Date(ddmmyyy2mmddyyyy(readableDate))

  return checkingDate < currentDate
}

export const getHighlightStatus = (
  active: boolean,
  days: string[],
): HighlightStatus => {
  if (days.every(isPastDate)) {
    return 'FINISHED'
  }

  if (!active) return 'PAUSED'
  if (days.includes(readableCurrentDate())) return 'RUNNING'
  return 'SCHEDULED'
}

const MONTH = DAYS_IN.MONTH * MILLISECONDS_IN.DAY
export const forwardOneMonth = () => new Date(+new Date() + MONTH)

export const SEPARATOR = ','
export const toggleJoinedDateString = ({
  toggleDate,
  joinedDateString,
}: {
  toggleDate: Date
  joinedDateString: string
}): string => {
  const readableDateSet = new Set(joinedDateString.split(SEPARATOR))
  const toggleDateString = dateToReadableStringDate(toggleDate)

  if (readableDateSet.has(toggleDateString)) {
    readableDateSet.delete(toggleDateString)
  } else {
    readableDateSet.add(toggleDateString)
  }

  return [...readableDateSet].join(SEPARATOR)
}
