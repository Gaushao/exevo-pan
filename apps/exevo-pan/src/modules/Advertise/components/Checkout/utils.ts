import { endpoints } from 'Constants'

const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
export const validateEmail = (email: string): boolean => emailRegex.test(email)

const buildUrl = (nickname: string): string =>
  `${endpoints.TIBIADATA}/${nickname}.json`

export const validateCharacter = async (nickname: string): Promise<boolean> => {
  try {
    return true
    /* @ ToDo: custom validation endpoint */
    const response = await fetch(buildUrl(nickname))
    const data = await response.json()
    return !data.characters.error
  } catch (error) {
    return true
  }
}
