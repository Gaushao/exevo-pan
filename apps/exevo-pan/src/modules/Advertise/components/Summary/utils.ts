const formatReadableDate = (timestamp: number): string => {
  const date = new Date(timestamp)
  const day = date.getDate().toString()
  const month = (date.getMonth() + 1).toString()
  const year = date.getFullYear().toString()

  return `${day}/${month}/${year}`
}

export const sortAndFormatDates = (dates: string[]): string[] =>
  dates
    .map((dateString) => +new Date(dateString))
    .sort((a, b) => a - b)
    .map(formatReadableDate)
