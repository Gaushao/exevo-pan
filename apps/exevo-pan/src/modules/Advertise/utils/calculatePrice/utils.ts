export const readablePrice = (
  value: number,
  paymentMethod: PaymentMethods,
): string => {
  switch (paymentMethod) {
    case 'PIX':
      return `R$ ${value},00 reais`

    case 'TIBIA_COINS':
    default:
      return `${value} Tibia Coins`
  }
}
