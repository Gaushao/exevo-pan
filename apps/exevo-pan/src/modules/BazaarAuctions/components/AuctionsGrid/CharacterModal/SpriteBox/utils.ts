export const addonCheck = {
  first: (type: number): boolean => type === 1 || type === 3,
  second: (type: number): boolean => type === 2 || type === 3,
}
