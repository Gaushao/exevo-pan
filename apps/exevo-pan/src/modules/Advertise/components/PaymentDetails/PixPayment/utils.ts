import { QrCodePix } from 'qrcode-pix'
import { advertising } from 'Constants'
import { calculatePrice } from '../../../utils'

interface PixParams {
  txId: number
  daysAmount: number
  isPro: boolean
}

export const generateQrCode = async ({
  txId,
  daysAmount,
  isPro,
}: PixParams): Promise<PixObject> => {
  const { totalPrice } = calculatePrice({
    days: daysAmount,
    paymentMethod: 'PIX',
    isPro,
  })
  const id = txId.toString()
  const tx = QrCodePix({
    version: '01',
    key: advertising.PIX_KEY,
    name: id,
    city: id,
    message: id,
    transactionId: id,
    value: totalPrice,
  })

  return {
    payload: tx.payload(),
    qrCode: await tx.base64(),
  }
}
