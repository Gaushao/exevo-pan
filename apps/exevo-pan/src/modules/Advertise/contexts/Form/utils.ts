import { dateToStandardStringDate } from 'utils'
import { getDatesUntilEnd } from 'components/Organisms/RangeDatePicker/utils'

const getLastItems = <T>(array: T[], amount: number): T[] =>
  array.slice(Math.max(array.length - amount, 0))

export const getRecommendedDays = (endDate: number): string[] =>
  getLastItems(getDatesUntilEnd(new Date(endDate * 1000)), 2).map(
    dateToStandardStringDate,
  )
