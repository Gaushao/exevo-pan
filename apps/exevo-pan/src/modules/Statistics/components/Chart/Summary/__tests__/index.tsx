import { screen } from '@testing-library/react'
import { renderWithProviders } from 'utils/test'
import Summary from '..'

describe('<Summary />', () => {
  test('should render values correctly', () => {
    renderWithProviders(
      <Summary title="Total volume" value={1000} percentage={50} />,
    )

    expect(
      screen.getByRole('heading', {
        name: 'Total volume',
      }),
    ).toBeInTheDocument()

    expect(screen.getByText('1,000 TC')).toBeInTheDocument()
    expect(screen.getByText(/50\.00%/)).toBeInTheDocument()
  })

  test('should style trend icon red/green', () => {
    const { rerender } = renderWithProviders(
      <Summary title="Total volume" value={1000} percentage={50} />,
    )

    const textElement = screen.getByText(/50\.00%/)

    /* expect(textElement).toHaveStyle('color: #377712') */
    expect(textElement).toMatchSnapshot()
    /* expect(screen.getByLabelText('Upwards trend')).toHaveStyle('fill: #377712') */
    expect(screen.getByLabelText('Upwards trend')).toMatchInlineSnapshot(`
      <svg
        aria-label="Upwards trend"
        class="mr-1 h-4 w-4 fill-green"
        height="24px"
        style="transform: none;"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
        />
      </svg>
    `)

    rerender(
      <Summary
        title="Total volume"
        value={1000}
        percentage={50}
        positive={false}
      />,
    )

    /* expect(textElement).toHaveStyle('color: #C51313') */
    expect(textElement).toMatchSnapshot()
    /* expect(screen.getByLabelText('Downwards trend')).toHaveStyle(
      'fill: #C51313',
    ) */
    expect(screen.getByLabelText('Downwards trend')).toMatchSnapshot()

    rerender(
      <Summary title="Total volume" value={1000} percentage={50} positive />,
    )

    /* expect(textElement).toHaveStyle('color: #377712') */
    expect(textElement).toMatchSnapshot()
    /* expect(screen.getByLabelText('Upwards trend')).toHaveStyle('fill: #377712') */
    expect(screen.getByLabelText('Upwards trend')).toMatchSnapshot()
  })
})
