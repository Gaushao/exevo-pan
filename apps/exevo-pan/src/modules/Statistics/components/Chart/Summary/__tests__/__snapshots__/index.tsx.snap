// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> should style trend icon red/green 1`] = `
<span
  class="text-tsm flex items-center font-bold text-green"
>
  <svg
    aria-label="Upwards trend"
    class="mr-1 h-4 w-4 fill-green"
    style="transform: none;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
    />
  </svg>
  50.00%
</span>
`;

exports[`<Summary /> should style trend icon red/green 3`] = `
<span
  class="text-tsm flex items-center font-bold text-red"
>
  <svg
    aria-label="Downwards trend"
    class="mr-1 h-4 w-4 fill-red"
    style="transform: scaleY(-1);"
    viewBox="0 0 24 24"
  >
    <path
      d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
    />
  </svg>
  50.00%
</span>
`;

exports[`<Summary /> should style trend icon red/green 4`] = `
<svg
  aria-label="Downwards trend"
  class="mr-1 h-4 w-4 fill-red"
  style="transform: scaleY(-1);"
  viewBox="0 0 24 24"
>
  <path
    d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
  />
</svg>
`;

exports[`<Summary /> should style trend icon red/green 5`] = `
<span
  class="text-tsm flex items-center font-bold text-green"
>
  <svg
    aria-label="Upwards trend"
    class="mr-1 h-4 w-4 fill-green"
    style="transform: none;"
    viewBox="0 0 24 24"
  >
    <path
      d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
    />
  </svg>
  50.00%
</span>
`;

exports[`<Summary /> should style trend icon red/green 6`] = `
<svg
  aria-label="Upwards trend"
  class="mr-1 h-4 w-4 fill-green"
  style="transform: none;"
  viewBox="0 0 24 24"
>
  <path
    d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
  />
</svg>
`;
